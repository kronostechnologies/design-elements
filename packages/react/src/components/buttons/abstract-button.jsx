import styled from 'styled-components';

import abstractStyle from './styles/abstract';

const AbstractButton = styled.button`
  ${abstractStyle}
`;

export default AbstractButton;
