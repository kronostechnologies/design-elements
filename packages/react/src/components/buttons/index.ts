export { Size } from './abstract/types';
export { ButtonProps, ButtonType, Type } from './types';
export { getBaseButtonStyles } from './abstract/styled';
export { getButtonTypeStyles, getButtonStyles } from './styled';
export { AbstractButton } from './abstract/abstract-button';
export { Button } from './button';
export { SearchButton } from './search-button';
export { IconButton } from './icon-button';
