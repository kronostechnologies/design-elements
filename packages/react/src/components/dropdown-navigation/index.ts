export { DropdownNavigation } from './dropdown-navigation';
export type { DropdownNavigationProps } from './dropdown-navigation';
