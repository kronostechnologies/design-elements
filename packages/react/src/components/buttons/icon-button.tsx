import { Theme } from '@design-elements/themes/theme';
import { focus } from '@design-elements/utils/css-state';
import React, { ReactElement } from 'react';
import styled from 'styled-components';

import { useDeviceContext } from '../device-context-provider/device-context-provider';
import { Icon, IconName } from '../icon/icon';
import { AbstractButton, AbstractButtonProps } from './abstract-button';

type ButtonType = 'primary' | 'secondary' | 'tertiary';
type Type = 'submit' | 'button' | 'reset';

interface ButtonProps {
    /**
     * Visual style
     * @default primary
     */
    buttonType: ButtonType;
    /**
     * Disables button
     * @default false
     */
    disabled?: boolean;
    /**
     * Name of the desired icon (refer to icon library)
     */
    iconName: IconName;
    /**
     * Sets aria-label
     */
    label: string;
    /**
     * Sets button type
     * @default submit
     */
    type?: Type;

    onClick?(): void;
}

interface StyledButtonProps {
    isMobile: boolean;
    theme: Theme;
    buttonType: ButtonType;
}

const StyledButton = styled(AbstractButton)<AbstractButtonProps & StyledButtonProps>`
    ${(props) => focus(props, true)};
    ${({ theme, buttonType }) => {
        switch (buttonType) {
            case 'primary':
                return `
                    background-color: ${theme.main['primary-1.1']};
                    border-color: ${theme.main['primary-1.1']};
                    color: ${theme.greys.white};

                    &:hover {
                        background-color: ${theme.main['primary-1.3']};
                    }

                    &:disabled {
                        background-color: ${theme.main['primary-1.2']};
                    }
                `;
            case 'secondary':
                return `
                    background-color: transparent;
                    border-color: ${theme.main['primary-1.1']};
                    color: ${theme.main['primary-1.1']};

                    &:hover {
                        border-color: ${theme.main['primary-1.3']};
                        color: ${theme.main['primary-1.3']};
                    }

                    &:disabled {
                        border-color: ${theme.main['primary-1.2']};
                        color: ${theme.main['primary-1.2']};
                    }
                `;
            case 'tertiary':
                return `
                    background-color: transparent;
                    border-color: transparent;
                    color: ${theme.greys['dark-grey']};

                    &:hover {
                        background-color: ${theme.greys.grey};
                        color: ${theme.greys.black};
                    }

                    &:disabled {
                        background-color: transparent;
                        color: ${theme.greys['mid-grey']};
                    }
                `;
        }
    }}

    padding: 0;
    width: ${({ isMobile }) => (isMobile ? '48px' : '32px')};
`;

export function IconButton({
    iconName,
    label,
    onClick,
    type = 'submit',
    buttonType,
    disabled,
    ...props
}: ButtonProps): ReactElement {
    const { isMobile } = useDeviceContext();
    const handleClick = (): void => onClick?.();

    return (
        <StyledButton
            aria-label={label}
            onClick={handleClick}
            isMobile={isMobile}
            type={type}
            buttonType={buttonType}
            disabled={disabled}
            {...props /* eslint-disable-line react/jsx-props-no-spreading */}
        >
            <Icon name={iconName} size={isMobile ? '20' : '16'} />
        </StyledButton>
    );
}
