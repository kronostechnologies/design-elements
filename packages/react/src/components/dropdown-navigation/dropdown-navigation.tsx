import {
    FunctionComponent,
    KeyboardEvent,
    MouseEvent as ReactMouseEvent, PropsWithChildren,
    ReactNode,
    useCallback,
    useEffect,
    useMemo,
    useRef,
    useState,
} from 'react';
import styled from 'styled-components';
import { useDataAttributes } from '../../hooks/use-data-attributes';
import { useTranslation } from '../../i18n/use-translation';
import { menuDimensions } from '../../legacy-constants/menuDimensions';
import { getRootDocument } from '../../utils/dom';
import { eventIsInside } from '../../utils/events';
import { v4 as uuid } from '../../utils/uuid';
import { Button } from '../buttons/button';
import { IconButton } from '../buttons/icon-button';
import { ButtonType } from '../buttons/types';
import { useDeviceContext } from '../device-context-provider/device-context-provider';
import { Icon, IconName } from '../icon/icon';
import { NavList } from '../nav-list/nav-list';
import { NavListOption } from '../nav-list/nav-list-option';

const StyledDiv = styled.div`
    position: relative;
`;

const StyledButton = styled(Button)<{ isMobile: boolean }>`
    font-size: 0.875rem;
    font-weight: var(--font-normal);
    text-transform: unset;
`;

const StyledRightIcon = styled(Icon)`
    margin-left: var(--spacing-1x);
`;

const StyledLeftIcon = styled(Icon)`
    margin-right: var(--spacing-1x);
`;

const StyledNavDropdown = styled(NavList)`
    max-width: ${menuDimensions.maxWidth};
    min-width: ${menuDimensions.minWidth};
    right: 0;
    width: initial;
`;

function getFirstFocusableElement(array: NavListOption[]): NavListOption {
    const focusableElements = array.filter((element) => !element.disabled);
    return focusableElements[0];
}

interface NavButtonProps {
    /**
     * Sets nav's description
     * @default 'Menu'
     * */
    ariaLabel?: string;
    autofocus?: boolean;
    tag?: 'div' | 'nav';
    buttonAriaLabel?: string;
    children?: ReactNode;
    className?: string;
    /**
     * Sets menu open by default
     * @default false
     * */
    defaultOpen?: boolean;
    /**
     * Sets chevron icon
     * @default true
     * */
    hasCaret?: boolean;
    iconName?: IconName;
    iconOnly?: boolean;
    id?: string;
    options: NavListOption[];
    title?: string;
    buttonType?: ButtonType;
    inverted?: boolean;
    onDropdownVisibilityChanged?(isOpen: boolean): void;
    onLinkSelected?(option: NavListOption): void;
}

export const DropdownNavigation: FunctionComponent<PropsWithChildren<NavButtonProps>> = ({
    tag,
    ariaLabel,
    autofocus,
    buttonAriaLabel,
    buttonType = 'tertiary',
    children,
    className,
    defaultOpen = false,
    hasCaret = true,
    iconName,
    iconOnly = false,
    id: providedId,
    inverted = true,
    onLinkSelected,
    onDropdownVisibilityChanged,
    options,
    title,
    ...props
}) => {
    const { isMobile } = useDeviceContext();
    const { t } = useTranslation('dropdown-navigation');
    const id = useMemo(() => providedId || uuid(), [providedId]);
    const [focusedValue, setFocusedValue] = useState('');
    const [isOpen, setOpen] = useState(defaultOpen);
    const containerAriaLabel = tag === 'div' ? '' : ariaLabel || t('ariaLabel');
    const dataAttributes = useDataAttributes(props);

    useEffect(() => {
        onDropdownVisibilityChanged?.(isOpen);
    }, [isOpen, onDropdownVisibilityChanged]);

    const buttonRef = useRef<HTMLButtonElement>(null);
    const NavListRef = useRef<HTMLUListElement>(null);
    const navRef = useRef<HTMLDivElement>(null);

    const handleClickOutside: (event: MouseEvent) => void = useCallback((event) => {
        const clickIsOutside = !eventIsInside(event, buttonRef.current, NavListRef.current);
        const shouldClose = (NavListRef.current === null || clickIsOutside) && isOpen;

        if (shouldClose) {
            setOpen(false);
        }
    }, [isOpen]);

    useEffect(() => {
        if (!isOpen) {
            setFocusedValue('');
        }

        document.addEventListener('mouseup', handleClickOutside);

        return () => {
            document.removeEventListener('mouseup', handleClickOutside);
        };
    }, [handleClickOutside, isOpen]);

    const handleNavDropdownKeyDown = (event: KeyboardEvent<HTMLInputElement>): void => {
        if (event.key === 'Escape') {
            setOpen(false);
            buttonRef.current?.focus();
        }

        if (isOpen) {
            setTimeout(() => {
                const focusedElement = getRootDocument(navRef.current)?.activeElement;
                const isFocusInsideNav = navRef.current?.contains(focusedElement || null);

                if (!isFocusInsideNav) {
                    setOpen(false);
                }
            });
        }
    };

    const handleOnLinkSelected: (option: NavListOption) => void = (option: NavListOption) => {
        onLinkSelected?.(option);
        setOpen(false);
    };

    const handleButtonClick = (event: ReactMouseEvent<HTMLButtonElement>): void => {
        const isKeyboardActivated = event.detail === 0;

        if (isKeyboardActivated) {
            setTimeout(() => {
                const firstFocusableElement = getFirstFocusableElement(options);
                setFocusedValue(firstFocusableElement.value);
            });
        }
        setOpen(!isOpen);
    };

    return (
        <StyledDiv
            data-testid="nav-container"
            as={tag}
            ref={navRef}
            className={className}
            id={id}
            aria-label={containerAriaLabel}
        >
            {!iconOnly && (
                <StyledButton
                    aria-label={buttonAriaLabel}
                    aria-expanded={isOpen}
                    autofocus={autofocus}
                    data-testid="navigation-button"
                    isMobile={isMobile}
                    onClick={handleButtonClick}
                    ref={buttonRef}
                    title={title}
                    type="button"
                    buttonType={buttonType}
                    inverted={inverted}
                    // eslint-disable-next-line react/jsx-props-no-spreading
                    {...dataAttributes}
                >
                    {iconName && <StyledLeftIcon aria-hidden="true" name={iconName} size="16" />}
                    {children}
                    {hasCaret && (
                        <StyledRightIcon
                            aria-hidden="true"
                            name={isOpen ? 'chevronUp' : 'chevronDown'}
                            size="16"
                        />
                    )}
                </StyledButton>
            )}
            {iconOnly && iconName && (
                <IconButton
                    aria-label={buttonAriaLabel}
                    aria-expanded={isOpen}
                    autofocus={autofocus}
                    data-testid="navigation-button"
                    iconName={iconName}
                    onClick={handleButtonClick}
                    ref={buttonRef}
                    title={title}
                    type="button"
                    buttonType={buttonType}
                    inverted={inverted}
                />
            )}
            <StyledNavDropdown
                data-testid="dropdown-navDropdown"
                focusedValue={focusedValue}
                onChange={handleOnLinkSelected}
                onKeyDown={handleNavDropdownKeyDown}
                options={options}
                ref={NavListRef}
                hidden={!isOpen}
            />
        </StyledDiv>
    );
};
