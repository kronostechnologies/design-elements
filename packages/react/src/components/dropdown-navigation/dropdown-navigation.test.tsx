import { shallow } from 'enzyme';
import { getByTestId } from '../../test-utils/enzyme-selectors';
import { mountWithProviders, renderWithProviders } from '../../test-utils/renderer';
import { IconButton } from '../buttons/icon-button';
import { DropdownNavigation } from './dropdown-navigation';

jest.mock('../../utils/uuid');

const options = [
    {
        label: 'Option A',
        value: 'optionA',
        href: '/testa',
    },
    {
        label: 'Option B',
        value: 'optionB',
        href: '/testb',
    },
    {
        label: 'Option C',
        value: 'optionC',
        href: '/testc',
    },
    {
        label: 'Option D',
        value: 'optionD',
        href: '/testd',
    },
];

const buttonTypes = ['normal', 'iconOnly'] as const;

describe('DropdownNavigation', () => {
    buttonTypes.forEach((type) => {
        const isIconOnly = type === 'iconOnly';

        test(`Adds aria-label to menu-button when buttonAriaLabel is defined (${type})`, () => {
            const ariaLabel = 'test-aria-label';

            const wrapper = shallow(
                <DropdownNavigation buttonAriaLabel={ariaLabel} options={options} iconOnly={isIconOnly} iconName="home">
                    Test Button
                </DropdownNavigation>,
            );

            expect(getByTestId(wrapper, 'menu-button').prop('aria-label')).toBe(ariaLabel);
        });

        test(`Opens nav-menu when menu-button is clicked (${type})`, () => {
            const wrapper = mountWithProviders(
                <DropdownNavigation options={options} iconOnly={isIconOnly} iconName="home">
                    Test Button
                </DropdownNavigation>,
            );

            getByTestId(wrapper, 'menu-button').simulate('click');

            expect(getByTestId(wrapper, 'menu-navMenu').prop('hidden')).toBe(false);
        });

        test(`Focuses the first menu-item when menu opens with Enter (${type})`, () => {
            const wrapper = mountWithProviders(
                <DropdownNavigation options={options} iconOnly={isIconOnly} iconName="home">
                    Test Button
                </DropdownNavigation>,
            );

            getByTestId(wrapper, 'menu-button').simulate('keydown', { key: 'Enter' });

            setTimeout(() => {
                expect(getByTestId(wrapper, 'menu-navMenu').prop('focusedValue')).toBe('optionA');
            });
        });

        test(`Focuses the first menu-item when menu opens with Space (${type})`, () => {
            const wrapper = mountWithProviders(
                <DropdownNavigation options={options} iconOnly={isIconOnly} iconName="home">
                    Test Button
                </DropdownNavigation>,
            );

            getByTestId(wrapper, 'menu-button').simulate('keydown', { key: ' ' });

            setTimeout(() => {
                expect(getByTestId(wrapper, 'menu-navMenu').prop('focusedValue')).toBe('optionA');
            });
        });

        test(`Focuses menu-button when escape key is pressed in nav-menu (${type})`, () => {
            const wrapper = mountWithProviders(
                <DropdownNavigation defaultOpen options={options} iconOnly={isIconOnly} iconName="home">
                    Test Button
                </DropdownNavigation>,
                { attachTo: document.body },
            );

            getByTestId(wrapper, 'listitem-optionA-link').simulate('keydown', { key: 'Escape' });

            expect(document.activeElement).toBe(getByTestId(wrapper, 'menu-button').getDOMNode());
            wrapper.unmount();
        });

        test(`Should call onMenuVisibilityChanged when nav-menu closes (${type})`, () => {
            const onMenuVisibilityChanged = jest.fn();
            const wrapper = mountWithProviders(
                <DropdownNavigation
                    defaultOpen
                    options={options}
                    iconOnly={isIconOnly}
                    iconName="home"
                    onMenuVisibilityChanged={onMenuVisibilityChanged}
                >
                    Test Button
                </DropdownNavigation>,
            );

            getByTestId(wrapper, 'menu-button').simulate('click');

            expect(onMenuVisibilityChanged).toHaveBeenCalledWith(false);
        });

        test(`Should call onMenuVisibilityChanged when nav-menu opens (${type})`, () => {
            const onMenuVisibilityChanged = jest.fn();
            const wrapper = mountWithProviders(
                <DropdownNavigation
                    options={options}
                    iconOnly={isIconOnly}
                    iconName="home"
                    onMenuVisibilityChanged={onMenuVisibilityChanged}
                >
                    Test Button
                </DropdownNavigation>,
            );

            getByTestId(wrapper, 'menu-button').simulate('click');

            expect(onMenuVisibilityChanged).toHaveBeenCalledWith(true);
        });
    });

    test('Should use IconButton component when iconOnly is true', () => {
        const wrapper = shallow(
            <DropdownNavigation options={options} iconOnly iconName="home">
                Test Button
            </DropdownNavigation>,
        );

        expect(wrapper.find(IconButton).exists()).toBe(true);
    });

    test('nav-menu is open when defaultOpen prop is set to true', () => {
        const wrapper = shallow(
            <DropdownNavigation defaultOpen options={options}>
                Test Button
            </DropdownNavigation>,
        );

        expect(getByTestId(wrapper, 'menu-navMenu').prop('hidden')).toBe(false);
    });

    test('Should close nav-menu when escape key is pressed in nav-menu', () => {
        const wrapper = mountWithProviders(
            <DropdownNavigation defaultOpen options={options}>
                Test Button
            </DropdownNavigation>,
        );

        getByTestId(wrapper, 'listitem-optionA-link').simulate('keydown', { key: 'Escape' });

        expect(getByTestId(wrapper, 'menu-navMenu').prop('hidden')).toBe(true);
    });

    test('Should call onMenuOptionsSelected when an option is selected in the nav-menu', () => {
        const onMenuOptionSelected = jest.fn();
        const wrapper = mountWithProviders(
            <DropdownNavigation options={options} onMenuOptionSelected={onMenuOptionSelected}>
                Test Button
            </DropdownNavigation>,
        );

        const navMenuOption = getByTestId(wrapper, `listitem-${options[0].value}-link`);
        navMenuOption.simulate('click');

        expect(onMenuOptionSelected).toHaveBeenCalledWith(expect.objectContaining(options[0]));
    });

    test('Matches Snapshot', () => {
        const tree = renderWithProviders(
            <DropdownNavigation options={options}>
                Test Button
            </DropdownNavigation>,
        );

        expect(tree).toMatchSnapshot();
    });

    test('Renders div container tag when "tag" prop is set to div', () => {
        const wrapper = mountWithProviders(
            <DropdownNavigation tag="div" options={options}>Test Button</DropdownNavigation>,
        );

        const navMenuContainer = getByTestId(wrapper, 'navmenu-container');
        expect(navMenuContainer.prop('as')).toEqual('div');
    });

    test('Renders nav container tag when "tag" props is set to nav', () => {
        const wrapper = mountWithProviders(
            <DropdownNavigation tag="nav" options={options}>Test Button</DropdownNavigation>,
        );

        const navMenuContainer = getByTestId(wrapper, 'navmenu-container');
        expect(navMenuContainer.prop('as')).toEqual('nav');
    });

    test('Matches Snapshot (tag="nav")', () => {
        const tree = renderWithProviders(
            <DropdownNavigation tag="nav" options={options}>
                Test Button
            </DropdownNavigation>,
        );

        expect(tree).toMatchSnapshot();
    });

    test('Matches Snapshot (defaultOpen)', () => {
        const tree = renderWithProviders(
            <DropdownNavigation defaultOpen options={options}>
                Test Button
            </DropdownNavigation>,
        );

        expect(tree).toMatchSnapshot();
    });
});
