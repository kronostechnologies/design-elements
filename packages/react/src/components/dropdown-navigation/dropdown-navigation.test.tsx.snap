// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownNavigation Matches Snapshot (defaultOpen) 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c2 {
  background-color: transparent;
  border-color: transparent;
  color: #FFFFFF;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #84C6EA;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c2:hover,
.c2[aria-expanded='true'] {
  background-color: rgb(0 0 0 / 0.5);
  border-color: transparent;
  color: #FFFFFF;
}

.c2:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #006296;
}

.c8 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1B1C1E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875rem;
  line-height: 2rem;
  margin: 2px;
  overflow: hidden;
  padding: 0 var(--spacing-2x);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c7:focus {
  box-shadow: 0 0 0 0 transparent;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c7:hover {
  background-color: #DBDEE1;
  color: #000000;
}

.c7[disabled] {
  color: #B7BBC2;
  pointer-events: none;
}

.c5 {
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  list-style-type: none;
  margin: 0;
  overflow-y: auto;
  padding: 0;
  position: absolute;
  width: 100%;
}

.c0 {
  position: relative;
}

.c3 {
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  text-transform: unset;
}

.c4 {
  margin-left: var(--spacing-1x);
}

.c6 {
  max-width: 350px;
  min-width: 200px;
  right: 0;
  width: initial;
}

<div
  aria-label="ariaLabel"
  class="c0"
  data-testid="nav-container"
  id="uuid1"
>
  <button
    aria-expanded="true"
    class="c1 c2 c3"
    data-testid="navigation-button"
    type="button"
  >
    Test Button
    <svg
      aria-hidden="true"
      class="c4"
      color="currentColor"
      focusable="false"
      height="16"
      width="16"
    />
  </button>
  <ul
    class="c5 c6"
    data-testid="dropdown-navDropdown"
    id="uuid2"
  >
    <li
      data-testid="listitem-optionA"
    >
      <a
        class="c7"
        data-testid="listitem-optionA-link"
        href="/testa"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option A
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionB"
    >
      <a
        class="c7"
        data-testid="listitem-optionB-link"
        href="/testb"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option B
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionC"
    >
      <a
        class="c7"
        data-testid="listitem-optionC-link"
        href="/testc"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option C
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionD"
    >
      <a
        class="c7"
        data-testid="listitem-optionD-link"
        href="/testd"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option D
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`DropdownNavigation Matches Snapshot (tag="nav") 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c2 {
  background-color: transparent;
  border-color: transparent;
  color: #FFFFFF;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #84C6EA;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c2:hover,
.c2[aria-expanded='true'] {
  background-color: rgb(0 0 0 / 0.5);
  border-color: transparent;
  color: #FFFFFF;
}

.c2:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #006296;
}

.c8 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1B1C1E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875rem;
  line-height: 2rem;
  margin: 2px;
  overflow: hidden;
  padding: 0 var(--spacing-2x);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c7:focus {
  box-shadow: 0 0 0 0 transparent;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c7:hover {
  background-color: #DBDEE1;
  color: #000000;
}

.c7[disabled] {
  color: #B7BBC2;
  pointer-events: none;
}

.c5 {
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  list-style-type: none;
  margin: 0;
  overflow-y: auto;
  padding: 0;
  position: absolute;
  width: 100%;
}

.c0 {
  position: relative;
}

.c3 {
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  text-transform: unset;
}

.c4 {
  margin-left: var(--spacing-1x);
}

.c6 {
  max-width: 350px;
  min-width: 200px;
  right: 0;
  width: initial;
}

<nav
  aria-label="ariaLabel"
  class="c0"
  data-testid="nav-container"
  id="uuid1"
>
  <button
    aria-expanded="false"
    class="c1 c2 c3"
    data-testid="navigation-button"
    type="button"
  >
    Test Button
    <svg
      aria-hidden="true"
      class="c4"
      color="currentColor"
      focusable="false"
      height="16"
      width="16"
    />
  </button>
  <ul
    class="c5 c6"
    data-testid="dropdown-navDropdown"
    hidden=""
    id="uuid2"
  >
    <li
      data-testid="listitem-optionA"
    >
      <a
        class="c7"
        data-testid="listitem-optionA-link"
        href="/testa"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option A
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionB"
    >
      <a
        class="c7"
        data-testid="listitem-optionB-link"
        href="/testb"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option B
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionC"
    >
      <a
        class="c7"
        data-testid="listitem-optionC-link"
        href="/testc"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option C
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionD"
    >
      <a
        class="c7"
        data-testid="listitem-optionD-link"
        href="/testd"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option D
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`DropdownNavigation Matches Snapshot 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c2 {
  background-color: transparent;
  border-color: transparent;
  color: #FFFFFF;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #84C6EA;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c2:hover,
.c2[aria-expanded='true'] {
  background-color: rgb(0 0 0 / 0.5);
  border-color: transparent;
  color: #FFFFFF;
}

.c2:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #006296;
}

.c8 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #1B1C1E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.875rem;
  line-height: 2rem;
  margin: 2px;
  overflow: hidden;
  padding: 0 var(--spacing-2x);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c7:focus {
  box-shadow: 0 0 0 0 transparent;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c7:hover {
  background-color: #DBDEE1;
  color: #000000;
}

.c7[disabled] {
  color: #B7BBC2;
  pointer-events: none;
}

.c5 {
  background-color: #FFFFFF;
  border: 1px solid #60666E;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  list-style-type: none;
  margin: 0;
  overflow-y: auto;
  padding: 0;
  position: absolute;
  width: 100%;
}

.c0 {
  position: relative;
}

.c3 {
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  text-transform: unset;
}

.c4 {
  margin-left: var(--spacing-1x);
}

.c6 {
  max-width: 350px;
  min-width: 200px;
  right: 0;
  width: initial;
}

<div
  aria-label="ariaLabel"
  class="c0"
  data-testid="nav-container"
  id="uuid1"
>
  <button
    aria-expanded="false"
    class="c1 c2 c3"
    data-testid="navigation-button"
    type="button"
  >
    Test Button
    <svg
      aria-hidden="true"
      class="c4"
      color="currentColor"
      focusable="false"
      height="16"
      width="16"
    />
  </button>
  <ul
    class="c5 c6"
    data-testid="dropdown-navDropdown"
    hidden=""
    id="uuid2"
  >
    <li
      data-testid="listitem-optionA"
    >
      <a
        class="c7"
        data-testid="listitem-optionA-link"
        href="/testa"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option A
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionB"
    >
      <a
        class="c7"
        data-testid="listitem-optionB-link"
        href="/testb"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option B
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionC"
    >
      <a
        class="c7"
        data-testid="listitem-optionC-link"
        href="/testc"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option C
        </span>
      </a>
    </li>
    <li
      data-testid="listitem-optionD"
    >
      <a
        class="c7"
        data-testid="listitem-optionD-link"
        href="/testd"
        tabindex="0"
      >
        <span
          class="c8"
        >
          Option D
        </span>
      </a>
    </li>
  </ul>
</div>
`;
